pkg/kubelet/kubelet.go

```diff
@@ -1789,6 +1789,20 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 		}
 	}
 
+	// handlePodResourcesResize updates the pod to use the allocated resources. This should come
+	// before the main business logic of SyncPod, so that a consistent view of the pod is used
+	// across the sync loop.
+	if kuberuntime.IsInPlacePodVerticalScalingAllowed(pod) {
+		// Handle pod resize here instead of doing it in HandlePodUpdates because
```

Comment:
Do we lose anything by dropping the check for CouldHaveRunningContainers [below](https://github.com/kubernetes/kubernetes/pull/128269/files#diff-67dd9e8f3ee257072765326cb4f242852554a2c0753563fa51e292c0a63a7b94L1951)? (I have to take a close look at the updated handlePodResize logic tomorrow, but just confirming this is intentional)

Reply:
I think the check was needed before because `AllocatedResources` was only set in the container status for running containers. Now that we're pulling from the internal store directly, this is no longer needed. Removing this allows pods to be resized even with no containers running, which can also reduce the restarts or resize operations needed.